future/dryrun.go

```diff
@@ -0,0 +1,109 @@
+package future
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/algorand/go-algorand-sdk/client/v2/algod"
+	"github.com/algorand/go-algorand-sdk/client/v2/common/models"
+	"github.com/algorand/go-algorand-sdk/types"
+)
+
+// CreateDryrun creates a DryrunRequest object from a client and slice of SignedTxn objects and a default configuration
+// Passed in as a pointer to a DryrunRequest object to use for extra parameters
+func CreateDryrun(client algod.Client, txns []types.SignedTxn, dr *models.DryrunRequest) (drr models.DryrunRequest, err error) {
+	var (
+		apps   []types.AppIndex
+		assets []types.AssetIndex
+		accts  []types.Address
+	)
+
+	drr.Txns = txns
+
+	if dr != nil {
+		drr.LatestTimestamp = dr.LatestTimestamp
+		drr.Round = dr.Round
+		drr.ProtocolVersion = dr.ProtocolVersion
+		drr.Sources = dr.Sources
+	}
+
+	for _, t := range txns {
+		if t.Txn.Type != types.ApplicationCallTx {
+			continue
+		}
+
+		accts = append(accts, t.Txn.Sender)
+
+		accts = append(accts, t.Txn.Accounts...)
+		apps = append(apps, t.Txn.ForeignApps...)
+		assets = append(assets, t.Txn.ForeignAssets...)
+
+		if t.Txn.ApplicationID == 0 {
+			appId := 1 //TODO: make the magic number?
```

Comment:
nit: I don't have a strong feeling on this one, probably need Andrew and Jason to weigh in on this.

Reply:
okay, @jasonpaulos told me there is a `defaultAppIdx` in goal, which is here: https://github.com/algorand/go-algorand/blob/master/libgoal/libgoal.go#L1006.

It is applied in https://github.com/algorand/go-algorand/blob/master/libgoal/libgoal.go#L1066-L1090, which is `MakeDryrunStateGenerated`.

I think this applies to TS-implementation too? What do you think?

Reply:
I can set it, I don't _think_ its required to be anything but !0 but for uniformity I'll make it match

Reply:
TY!